de.dlr.sc.virsat.model.extension.tests.test/src/de/dlr/sc/virsat/model/extension/tests/model/json/TestCategoryCompositionArrayTest.java

```diff
@@ -0,0 +1,136 @@
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package de.dlr.sc.virsat.model.extension.tests.model.json;
+
+import static de.dlr.sc.virsat.model.extension.tests.test.TestActivator.assertEqualsNoWs;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.io.StringWriter;
+
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import javax.xml.bind.Unmarshaller;
+import javax.xml.transform.stream.StreamSource;
+
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.emf.ecore.resource.ResourceSet;
+import org.eclipse.emf.ecore.resource.impl.ResourceImpl;
+import org.eclipse.emf.ecore.resource.impl.ResourceSetImpl;
+import org.junit.Before;
+import org.junit.Test;
+
+import de.dlr.sc.virsat.model.concept.list.IBeanList;
+import de.dlr.sc.virsat.model.concept.types.property.BeanPropertyComposed;
+import de.dlr.sc.virsat.model.dvlm.concepts.Concept;
+import de.dlr.sc.virsat.model.dvlm.json.JAXBUtility;
+import de.dlr.sc.virsat.model.dvlm.types.impl.VirSatUuid;
+import de.dlr.sc.virsat.model.extension.tests.model.AConceptTestCase;
+import de.dlr.sc.virsat.model.extension.tests.model.TestCategoryAllProperty;
+import de.dlr.sc.virsat.model.extension.tests.model.TestCategoryCompositionArray;
+import de.dlr.sc.virsat.model.extension.tests.test.TestActivator;
+
+public class TestCategoryCompositionArrayTest extends AConceptTestCase {
+	
+	private TestCategoryCompositionArray testArray;
+	private JAXBUtility jaxbUtility;
+	private Concept concept;
+	private static final String RESOURCE = "/resources/json/TestCategoryCompositionArray_Marshaling.json";
+	private static final String RESOURCE_CHANGE_ELEMENT = "/resources/json/TestCategoryCompositionArray_Marshaling_ChangeElement.json";
+
+	@Before
+	public void setup() throws JAXBException {
+		// Because a composed bean is generic we need an adapter
+		// And because of the adapter we have to manually add the composed
+		// classes to the marshaller.
+		jaxbUtility = new JAXBUtility(new Class[] {TestCategoryCompositionArray.class, TestCategoryAllProperty.class});
+		
+		// Load the concept to create the test object
+		prepareEditingDomain();
+		concept = loadConceptFromPlugin();
+		testArray = new TestCategoryCompositionArray(concept);
+		
+		testArray.getATypeInstance().setUuid(new VirSatUuid("f5d016ac-65fa-4b9d-ae94-582d4f73138a"));
+		IBeanList<BeanPropertyComposed<TestCategoryAllProperty>> staticArray = testArray.getTestCompositionArrayStaticBean();
+		for (int i = 0; i < staticArray.size(); i++) {
+			String lastChar = Integer.toString(i);
```

Comment:
Any reason we cant just to `+ i` in the setUuid call? Java should convert the int to a string for you.